spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,92 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
```

Comment:
Why we are not including the insert?

Btw: Any change of JSON inclusion, I mean for `insert`:

```
INSERT INTO mytable JSON '{ "\"myKey\"": 0, "value": 0}';
```

I know that NewSQL and some relational databases have support for it, for example, Postgresql, but I am not sure about this on JPQL.

Reply:
JPQL (at least [version 3.1](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1#bnf) of it) does not have `insert`.  In the future if we want to add the ability to insert from JSON, hopefully we can write a better API for that rather than having the user write query language for it.

Reply:
What do you think about removing any data alteration for this version, keeping only `select`?

Reply:
> Why we are not including the `insert`?

The `insert ... values ....` form tends not to be very useful, since it competes directly with `@Insert`.

On the other hand, `insert ... select ....` is useful in very specific situations, but JPQL doesn't have it, and I doubt it can be implemented on most non-relational datastores.

HQL _does_ have both those things, but I really don't see a lot of pressure for them to be incorporated into the spec.

> Any change of JSON inclusion

I would not have thought that _literal_ JSON within the query itself was very useful. (But I might be wrong.)

What we support in Hibernate is:

- mapping a field of type `String` to a column of type `JSON`, or
- mapping an embeddable to a database column of type `JSON`.

But then the fact that this piece of an entity is stored as JSON is really just a question for the "mapping layer", and not something we need to address in the spec (given the current goals of the spec).

Reply:
> What do you think about removing any data alteration for this version, keeping only `select`?

I think this should be at least a minimal superset of the functionality you had already defined with the magic method name query stuff.

And the truth is that `update` and `delete` queries are very useful, because what they do _cannot_ be accomplished via `@Update` and `@Delete`.